lms/security.py

```diff
@@ -66,47 +67,60 @@ def get_route_policy(self, request):
         return self.fallback_policy()
 
 
-class LTISecurityPolicy:
+class _LTIUserBasedSecurityPolicy:
     @classmethod
-    def authenticated_userid(cls, request):
-        if request.lti_user is None:
-            return None
+    def get_lti_user(cls, request) -> LTIUser:
+        raise NotImplemented()
+
+    def permits(self, request, context, permission):
+        return _permits(self, request, context, permission)
+
+    def remember(self, request, userid, **kw):
+        pass
+
+    def forget(self, request):
+        pass
 
+    @staticmethod
+    def get_userid(lti_user):
         # urlsafe_b64encode() requires bytes, so encode the userid to bytes.
-        user_id_bytes = request.lti_user.user_id.encode()
+        user_id_bytes = lti_user.user_id.encode()
 
         safe_user_id_bytes = base64.urlsafe_b64encode(user_id_bytes)
 
         # urlsafe_b64encode() returns ASCII bytes but we need unicode, so
         # decode it.
         safe_user_id = safe_user_id_bytes.decode("ascii")
 
-        return ":".join([safe_user_id, str(request.lti_user.application_instance_id)])
+        return ":".join([safe_user_id, str(lti_user.application_instance_id)])
 
-    def identity(self, request):
-        userid = self.authenticated_userid(request)
+    def authenticated_userid(self, request):
```

Comment:
Now the value it's taken from identity directly. This simplifies things moving forward as only of of `authenticated_userid` / `identity` needs to build `LTIUser`.